src/main/java/is/citizen/sdk/api/CitizenApi.java

```diff
@@ -110,6 +112,84 @@ private void log(int status, String errorMessage) {
         return Optional.empty();
     }
 
+    /**
+     * Convert an authentication public key from the Citizen Service into a Java {@link PublicKey}
+     *
+     * @param publicKeyString authentication public key in Base 64 encoded DER format.
+     *
+     * @return {@link PublicKey} converted public key
+     */
+    public Optional<PublicKey> convertAuthPublicKey(String publicKeyString) {
+
+        try {
+            PublicKey publicKey = citizenCrypto.getPublicAuthKeyFromEncodedString(publicKeyString);
+            return Optional.of(publicKey);
+        } catch (CryptoException e) {
+            log(Constant.CITIZEN_CRYPTO_ERROR, getStackTrace(e));
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * Convert an encrypted authentication private key from the Citizen Service into a Java {@link PrivateKey}
+     *
+     * @param privateKeyString encrypted authentication private key.
+     * @param password password for the encrypted private key
+     *
+     * @return {@link PrivateKey} converted private key
```

Comment:
or empty if the password is incorrect or key is invalid?

Reply:
Fixed.